scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -937,9 +937,17 @@ class TypedPipeInst[T] private[scalding] (@transient inpipe: Pipe,
       // for us. So unwind until you hit the first filter, snapshot,
       // then apply the unwound functions
       case Some((tap, fields, Converter(conv))) =>
-        Execution.from(new Iterable[T] {
-          def iterator = m.openForRead(conf, tap).asScala.map(tup => conv(tup.selectEntry(fields)))
-        })
+        /*
+           * We create a Stream here so we are lazy with construction, but we don't read twice.
+           */
+        val tryStream = scala.util.Try(m.openForRead(conf, tap)
+          .asScala
+          .map(tup => conv(tup.selectEntry(fields)))
+          .toStream)
```

Comment:
How big are the things your reading avi that putting it in the job conf is bad? I guess if your concerned with filling up memory quite large. We are running into the issue of possibly reading it multiple times. But also the previous step can have 100's to 1000's of part files, each being read by a followup set of 1000's to 10,000's of mappers. So the number of read operations even just reading it once per mapper ends up being huge. Our SRE's are not thrilled :)